packages/components/src/internal/util/Date.ts

```diff
@@ -97,6 +97,13 @@ export function getUnFormattedNumber(n): number {
     return n ? numeral(n).value() : n;
 }
 
+// Issue 44398: see DateUtil.java getJsonDateTimeFormatString(), this function is to match the format, which is
+// provided by the LabKey server for the API response, from a JS Date object
+export function getJsonDateTimeFormatString(date: Date): string {
+    if (!date) return null;
```

Comment:
I assume `null` is what is wanted instead of `undefined` so the value doesn't get removed when posting to the server.  If so, a comment may be called for.

Reply:
I was just matching the `null` return case from the `parseDate`, but I see now that the `_formatDate` function is returning `undefined` so this check in `getJsonDateTimeFormatString` probably isn't needed. I'll verify the "value doesn't get removed when posting to the server" case and update accordingly here.

Reply:
It's possible parseDate was returning null because the way the date inputs behave. IIRC react-datepicker needs null  values not undefined, or maybe it just differentiates behavior based on null/undefined. I can't remember exactly but I do recall having issues around date inputs and null/undefined pretty recently.